client/components/subscriptionsPage/Subscriptions.jsx

```diff
@@ -1,21 +1,52 @@
-import React from 'react';
+import React, { useContext, useState } from 'react';
 import { Container, Row, Col, Card, Button, Badge } from 'react-bootstrap';
 import SubscriptionCard from './SubscriptionCard.jsx';
 import specifications from './Specifications.js';
+import { AuthContext } from '../../context/authContext.jsx';
+import axios from 'axios';
+import { Redirect } from 'react-router-dom';
+import { Alert } from 'react-bootstrap';
 
 export default function Subscriptions() {
-  //Need a handlesubscribe function
-  const handleSubscribe = () => {};
+  const user = useContext(AuthContext);
+  const [redirect, setRedirect] = useState(false);
+  const [error, setError] = useState('');
 
-  return (
+  const handleSubscribe = (idx) => {
+    if (!user.payment) {
+      setError('Please add payment method first in setting page');
+    } else {
+      axios
+        .post('/user/subscription', {
+          userId: user._id,
+          name: specifications[idx].name,
+          description: specifications[idx].description,
+          voucher: idx === 0 ? 1 : idx === 1 ? 3 : 5
+        })
+        .then(() => {
+          setRedirect(true);
+        })
+        .catch(() => console.log('OPPS'));
```

Comment:
I would at the very least console log the error hah

Reply:
Okay, will update when free.

Reply:
I like OOPS